### PR TITLE
First uncertainty model

### PR DESCRIPTION
**Pull request by marcinkusz**
Implements first centerpoint object detector with uncertaitny estimtation:

- Implements Gaussian NLL loss
- Adds centerpoint head with uncertainty estimation capabilities
- Extends demo script to visualize the 95% Gaussian CI for debugging

### CODE REVIEW

File: pcdet/utils/loss_utils.py [hunk 1]
```diff
@@ -344,6 +345,50 @@ def forward(self, out, target, mask=None):
...
+    num = mask.float().sum()
+    mask = mask.unsqueeze(2).expand_as(target).float()
+    isnotnan = (~ torch.isnan(target)).float()
+    mask *= isnotnan
+
+    pred = pred * mask
+    target = target * mask
+    sigma = sigma * mask
+
+    loss =  torch.abs(pred - target)**2 # was L1 in mmdet3d i.e. torch.abs(pred - target)
```
Comment (corneliusschroeder) [inline pcdet/utils/loss_utils.py:379]: is this correct the correct euclidean distance between two vectors?

Comment (marcinkusz) [reply pcdet/utils/loss_utils.py:379]: No, but it does correspond to the gaussian NLL loss
$L_{reg}=\frac{1}{2}exp(-\lambda^T)(v_{gt} - v)^2+\frac{1}{2}\lambda^T\mathbf{1}$ if we interpret the model output as the log variance i.e. $\lambda = log(\sigma^2)$ or [the pytorch implementation](https://pytorch.org/docs/stable/generated/torch.nn.GaussianNLLLoss.html) if we say that the model approximates the variance directly.
